compiler-cli/src/dependencies.rs

```diff
@@ -298,7 +298,7 @@ impl LocalPackages {
             .collect();
         self.packages
             .iter()
-            .filter(|(n, v)| !manifest_packages.contains(&(&SmolStr::from(n), v)))
+            .filter(|(n, v)| !manifest_packages.contains(&(&EcoString::inline(n), v)))
```

Comment:
The inline method will panic if the string is too large to be lined, so we can't use it with anything but a string literal.

I think we should always use the Into trait maybe?

Reply:
Ah, good catch! 

Got rid of all uses of `EcoString::inline`, so please have a look at my last commit. If it looks good I will go ahead and squash it :blush: